File: Problem5_Tokyo/5_claude_sol2.py
import sys


def solve():
    data = sys.stdin.buffer.read().split()
    pos = 0

    def read_token():
        nonlocal pos
        v = data[pos]
        pos += 1
        return v

    T = int(read_token())
    results = []

    for _ in range(T):
        N = int(read_token())

        prefix = [0] * (N + 1)
        for i in range(1, N + 1):
            direction = read_token()
            angle = int(read_token())
            step = angle if direction == b'L' else -angle
            prefix[i] = prefix[i - 1] + step

        # order indices by (prefix value, index) so equal-value groups are
        # contiguous and internally sorted by index
        order = sorted(range(N + 1), key=lambda k: (prefix[k], k))

        best_len = 0
        best_s = 0
        best_e = 0

        i = 0
        total = N + 1
        while i < total:
            j = i
            val = prefix[order[i]]
            while j < total and prefix[order[j]] == val:
                j += 1
            min_idx = order[i]
            max_idx = order[j - 1]
            length = max_idx - min_idx
            if length > 0:
                if length > best_len or (length == best_len and (min_idx + 1) < best_s):
                    best_len = length
                    best_s = min_idx + 1
                    best_e = max_idx
            i = j

        if best_len == 0:
            results.append("0 0 0")
        else:
            results.append(f"{best_len} {best_s} {best_e}")

    sys.stdout.write("\n".join(results) + "\n")


solve()

File: Problem5_Tokyo/5_claude_sol2.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        StreamTokenizer st = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));

        st.nextToken();
        int T = (int) st.nval;
        StringBuilder sb = new StringBuilder();

        for (int t = 0; t < T; t++) {
            st.nextToken();
            int N = (int) st.nval;

            long[] prefix = new long[N + 1];
            for (int i = 1; i <= N; i++) {
                st.nextToken();
                String dir = st.sval;
                st.nextToken();
                long x = (long) st.nval;
                prefix[i] = prefix[i - 1] + (dir.equals("L") ? x : -x);
            }

            Integer[] order = new Integer[N + 1];
            for (int i = 0; i <= N; i++)
                order[i] = i;

            final long[] pf = prefix;
            Arrays.sort(order, (a, b) -> {
                if (pf[a] != pf[b])
                    return Long.compare(pf[a], pf[b]);
                return Integer.compare(a, b);
            });

            long bestLen = 0;
            int bestS = 0, bestE = 0;

            int i = 0;
            while (i <= N) {
                int j = i;
                long val = pf[order[i]];
                while (j <= N && pf[order[j]] == val)
                    j++;

                int minIdx = order[i];
                int maxIdx = order[j - 1];
                long len = (long) maxIdx - minIdx;

                if (len > 0) {
                    if (len > bestLen || (len == bestLen && (minIdx + 1) < bestS)) {
                        bestLen = len;
                        bestS = minIdx + 1;
                        bestE = maxIdx;
                    }
                }
                i = j;
            }

            if (bestLen == 0) {
                sb.append("0 0 0\n");
            } else {
                sb.append(bestLen).append(' ').append(bestS).append(' ').append(bestE).append('\n');
            }
        }

        System.out.print(sb);
    }
}